File: backend/src/main/scala/order/api/SubmitPartialRefundRequestApi.scala
package order.api

import domain.shared.given

import admin.app.*
import cats.effect.IO
import domain.order.SubmitPartialRefundRequest
import domain.shared.{DeliveryAppState, OrderId, UserRole}
import org.http4s.HttpRoutes
import org.http4s.circe.CirceEntityCodec.*
import org.http4s.dsl.io.*
import shared.api.routing.*
import shared.app.*

val submitPartialRefundRequestApi: FixedMethodApi1[OrderId, DeliveryAppState] =
  jsonPostApi1[OrderId, SubmitPartialRefundRequest, DeliveryAppState](
    List(routeSegment("api"), routeSegment("delivery"), routeSegment("orders")),
    List(routeSegment("partial-refunds")),
  )

File: frontend/src/apis/order/SubmitPartialRefundRequestApi.ts
import type { DeliveryAppState, OrderId, SubmitPartialRefundRequest } from '@/objects/core/SharedObjects'
import { postNormalizedDeliveryState } from '@/system/api/DeliveryStateClient'
import { buildApiPath1, defineJsonPostApi1, routeSegment } from '@/system/api/TypedApiDefinitions'

export const submitPartialRefundRequestApiDefinition = defineJsonPostApi1<OrderId, SubmitPartialRefundRequest, DeliveryAppState>(
    [routeSegment('api'), routeSegment('delivery'), routeSegment('orders')],
    [routeSegment('partial-refunds')],
  )

export function submitPartialRefundRequest(
  orderId: OrderId,
  payload: SubmitPartialRefundRequest,
) {
  return postNormalizedDeliveryState(buildApiPath1(submitPartialRefundRequestApiDefinition, orderId), payload)
}
